File: Combination Sum.cpp


class Solution {
public:
       
        void cs(int index,vector<int>& candidates,vector<vector<int>>& ans,int target,vector<int> ds){

            if(index==candidates.size()){
                if(target==0){
                    ans.push_back(ds);
                }
                return;
            }

             if(candidates[index]<=target){
            ds.push_back(candidates[index]);
            cs(index,candidates,ans,target-candidates[index],ds);
            ds.pop_back();
             }

            cs(index+1,candidates,ans,target,ds);


        }

    vector<vector<int>> combinationSum(vector<int>& candidates, int target) {
        vector<vector<int>> ans;
        vector<int> ds;
        cs(0,candidates,ans,target,ds);
        return ans;
        
    }
};

File: Combination_Sum.js
function combination (candidates, target, index, sum, ans){

    if(target == 0){
        ans.push([...sum]);
        return ;
    }
    if(index == candidates.length || target < candidates[index]){
        return;
    }
    sum.push(candidates[index]);
    combination (candidates, target-candidates[index], index, sum, ans)
     sum.pop();
    combination (candidates, target, index + 1, sum, ans)

 }
var combinationSum = function(candidates, target) {
 let ans  = []   ;
 candidates.sort((a,b) => a - b );
combination(candidates, target, 0, [], ans )
return ans ;
    
};
